src/XMakeBuildEngine/Evaluation/ProjectRootElementCache.cs

```diff
@@ -194,7 +194,7 @@ internal ProjectRootElementCache(bool autoReloadFromDisk)
         /// <param name="preserveFormatting"><code>true</code> to the project was loaded with the formated preserved, otherwise <code>false</code>.</param>
         /// <returns>The ProjectRootElement instance if one exists.  Null otherwise.</returns>
         internal ProjectRootElement Get(string projectFile, OpenProjectRootElement openProjectRootElement, bool isExplicitlyLoaded,
-            bool preserveFormatting)
+            bool? preserveFormatting)
```

Comment:
At some point in the ProjectRootElement construction chain something has to choose either true or false for `PreserverFormatting`, otherwise the nullable bool flows all the way into the ProjectRootElement constructor (via very ambiguous detours through the cache callbacks). It gets interesting how the constructor should interpret the null value, maybe leave it off to the `System.Xml` default?

I've started thinking that we shouldn't expose format preservation in the cache, and instead change the ProjectRootElement such that you can mutate its `PreserveFormatting` property. Makes the Cache code easier, and follows the general pattern in the code that ProjectRootElement instances mutate and send change notifications rather than get evicted from the cache. The only cache eviction code is the `autoReloadFromDisk` feature, which is turned off either way (and maybe it should use Reload instead eviction :) ).